src/fscacher/cache.py

```diff
@@ -93,7 +93,10 @@ def fingerprinted(path, *args, **kwargs):
             parameters=tuple(sig.parameters.values()) + (fp_kwarg_param,)
         )
         fingerprinted.__signature__ = sig2
-        fingerprinted = self.memoize(fingerprinted)
+        # we need to ignore 'path' since we would like to dereference if symlink
+        # but then expect joblib's caching work on both original and dereferenced
+        # So we will add dereferenced path into fingerprint_kwarg
+        fingerprinted = self.memoize(fingerprinted, ignore=['path'])
```

Comment:
@yarikoptic This won't work if the decorated function's first argument is named something other than `path`.  Cf. #62.

Reply:
Also, you don't seem to have pre-commit installed in this repository.

Reply:
damn, I thought it would be of `fingerprinted` where **we** called it `path`.   

<details>
<summary>But making this change to tets</summary> 

```patch
diff --git a/src/fscacher/tests/test_cache.py b/src/fscacher/tests/test_cache.py
index 50ef10e..a7cc182 100644
--- a/src/fscacher/tests/test_cache.py
+++ b/src/fscacher/tests/test_cache.py
@@ -171,10 +171,10 @@ def test_memoize_path_dir(cache, tmp_path):
     calls = []
 
     @cache.memoize_path
-    def memoread(path, arg, kwarg=None):
-        calls.append([path, arg, kwarg])
+    def memoread(p, arg, kwarg=None):
+        calls.append([p, arg, kwarg])
         total_size = 0
-        with os.scandir(path) as entries:
+        with os.scandir(p) as entries:
             for e in entries:
                 if e.is_file():
                     total_size += e.stat().st_size
@@ -290,9 +290,9 @@ def test_memoize_path_tokens(tmp_path, cache, cache_tokens):
     calls = []
 
     @cache.memoize_path
-    def memoread(path, arg, kwarg=None):
-        calls.append(["cache", path, arg, kwarg])
-        with open(path) as f:
+    def memoread(p, arg, kwarg=None):
+        calls.append(["cache", p, arg, kwarg])
+        with open(p) as f:
             return f.read()
 
     @cache_tokens.memoize_path

```
</details>

breaks it indeed.  Could you please take over this PR (I need to switch, meeting is coming) and add `inspect.getargspec` to get actual name of the first argument of the original `f` instead of the harcoded `"path"`?

Reply:
Done.